src/Doctrine/Annotation/Tokens.php

```diff
@@ -272,7 +272,7 @@ public function offsetUnset($index): void
         }
 
         $max = \count($this) - 1;
-        while ($index < $max) {
+        while ($index < $max && isset($this[$index + 1])) {
```

Comment:
I'm not a fan of this type of change., From the app view and domain we know for sure that `$this[$index + 1]` is always set, so the check is overcomplete. I wouldn't change valid code just because some SCA tooling triggers a fails issue.

Reply:
Investigating this a bit more, it seems `Tokens` and/or `SplFixedArray` have some weird behavior I don't fully understand. I tried the following:
```php
$tokens = Tokens::fromCode('<?php exit;');
unset($tokens[0]);
var_export($tokens);
```
This is what I get:
```php
PhpCsFixer\Tokenizer\Tokens::__set_state(array(
   'blockStartCache' => 
  array (
  ),
   'blockEndCache' => 
  array (
  ),
   'codeHash' => '2321664371',
   'changed' => true,
   'foundTokenKinds' => 
  array (
    390 => 0,
    323 => 1,
    ';' => 1,
  ),
   0 => NULL,
   1 => 
  PhpCsFixer\Tokenizer\Token::__set_state(array(
     'content' => 'exit',
     'id' => 323,
     'isArray' => true,
     'changed' => false,
  )),
   2 => 
  PhpCsFixer\Tokenizer\Token::__set_state(array(
     'content' => ';',
     'id' => NULL,
     'isArray' => false,
     'changed' => false,
  )),
))
```
This is not what I expected when looking at the implementation of `Tokens::offsetUnset()`.

Then I updated that method:
```php
    public function offsetUnset($index): void
    {
        throw new LogicException('Not implemented.');
    }
```
Same result and all tests seem to pass 🤨️

Anyway, `SplFixedArray` seems to always allow `null` as value somehow, so I suppose the error reported by PHPStan should be considered; but since I don't really understand why yet, I'd rather ignore it for now. What do you think?

Reply:
I think you confused the `PhpCsFixer\Tokenizer\Tokens` with `PhpCsFixer\Doctrine\Annotation\Tokens` in your example, which I totally get as I thought we were looking at the first as well ;)

Reply:
>tests/Doctrine/Annotation/TokensTest.php

Is lacking a lot of test on the functionality the collection delivers to be fair.
I've added this (rough and ugly test) locally to it to test the upper boundary `unset` and it seems fine:

```php
    public function testCreateAndUnset(): void
    {
        $docComment = '/** @var';

        $token = new Token([T_DOC_COMMENT, $docComment.'*/']);
        $tokens = Tokens::createFromDocComment($token);

        static::assertCount(4, $tokens);

        unset($tokens[3]);

        static::assertCount(3, $tokens);
        static::assertSame($docComment, $tokens->getCode());
    }
```

Reply:
Well… that explains a lot 😂️

Change reverted, the error is now ignored.

Reply:
> > tests/Doctrine/Annotation/TokensTest.php
>
> Is lacking a lot of test on the functionality the collection delivers to be fair.

I noticed that as well. But since PHP now supports native attributes, I think Doctrine Annotations will become obsolete and I intend to deprecate all related fixers soon™, so maybe improving those tests is not worth the hassle. What do you think?

Reply:
Yeah I agree :+1: 
So extending the tests is no requirement in the PR what so ever. If someone would make a PR to add more coverage I would take a look for it, but if not.. well there are no open issues on it so all good :)